### ISSUE TITLE
Refactor Account Types

### ISSUE THREAD

**Issue by turbolent** [onflow/cadence#2641]
```[tasklist]
### Tasks
- [x] FLIP: https://github.com/onflow/flips/pull/92
- [ ] Implementation: https://github.com/onflow/cadence/pull/2648
```

**Comment by turbolent** [comment]
Blocked by #2643

**Comment by cybercent** [comment]
Is it feasible to include a fix such that `account.type(at: path)` does not crash when the path is set by an out-of-date contract, or is this out of scope?

**Comment by turbolent** [comment]
@cybercent That's a good idea, but should be a separate issue. Could you please open a feature request?

**Comment by cybercent** [comment]
Thanks, opened here:
https://github.com/onflow/cadence/issues/2715
